### PR TITLE
[license] Rename `@mui/x-license-pro` to `@mui/x-license`

### PR DESCRIPTION
**Pull request by cherniavskii**
Currently, `@mui/x-license-pro` is used for Pro **and** Premium packages.
This is confusing, so I'm taking the opportunity to make this change in v7 beta.

TODO:
- [x] Update migration guide

### Breaking changes

- If you're using the [commercial license](https://next.mui.com/x/introduction/licensing), you need to update the import path:

```diff
-import { LicenseInfo } from '@mui/x-license-pro';
+import { LicenseInfo } from '@mui/x-license';
```

### CODE REVIEW

File: docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md [hunk 1]
```diff
@@ -31,6 +35,22 @@ To have the option of using the latest API from `@mui/material`, the package pee
 It is a change in minor version only, so it should not cause any breaking changes.
 Please update your `@mui/material` package to this or a newer version.
 
+## Update the license package
+
+If you're using the commerical versions of the Data Grid ([Pro](/x/introduction/licensing/#pro-plan) and [Premium](/x/introduction/licensing/#premium-plan) plans), you need to update the license package to the latest version.
```
Comment (LukasTy) [inline docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md:40]: Are we sure about this? 🤔 
The license package is a direct dependency of the commercial packages, the latest version will be downloaded if they are using the latest package version. 🤔 
If we are referring to cases, where people are adding this dependency due to their package manager behavior, then maybe can make this section optional and move it below the import update, which is a clear BC for everyone? 🤔

**Review by LukasTy — COMMENTED** [review]
Great work! 👏

File: docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md [hunk 2]
```diff
@@ -31,6 +35,22 @@ To have the option of using the latest API from `@mui/material`, the package pee
...
+## Update the license package
+
+If you're using the commercial version of the Data Grid ([Pro](/x/introduction/licensing/#pro-plan) and [Premium](/x/introduction/licensing/#premium-plan) plans), you need to update the import path:
+
+```diff
+-import { LicenseInfo } from '@mui/x-license-pro';
++import { LicenseInfo } from '@mui/x-license';
+```
+
+If you have `@mui/x-license-pro` in the `dependencies` section of your `package.json`, update the license package to the latest version:
```
Comment (LukasTy) [inline docs/data/migration/migration-data-grid-v6/migration-data-grid-v6.md:47]:
```suggestion
If you have `@mui/x-license-pro` in the `dependencies` section of your `package.json`, rename and update the license package to the latest version:
```

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 3]
```diff
@@ -31,6 +31,22 @@ To have the option of using the latest API from `@mui/material`, the package pee
...
+## Update the license package
+
+If you're using the commercial version of the pickers ([Pro](/x/introduction/licensing/#pro-plan) plan), you need to update the import path:
+
+```diff
+-import { LicenseInfo } from '@mui/x-license-pro';
++import { LicenseInfo } from '@mui/x-license';
+```
+
+If you have `@mui/x-license-pro` in the `dependencies` section of your `package.json`, update the license package to the latest version:
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:43]:
```suggestion
If you have `@mui/x-license-pro` in the `dependencies` section of your `package.json`, rename and update the license package to the latest version:
```

File: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md [hunk 4]
```diff
@@ -31,6 +31,22 @@ To have the option of using the latest API from `@mui/material`, the package pee
 It is a change in minor version only, so it should not cause any breaking changes.
 Please update your `@mui/material` package to this or a newer version.
 
+## Update the license package
+
+If you're using the commercial version of the pickers ([Pro](/x/introduction/licensing/#pro-plan) plan), you need to update the import path:
```
Comment (LukasTy) [inline docs/data/migration/migration-pickers-v6/migration-pickers-v6.md:36]:
```suggestion
If you're using the commercial version of the Pickers ([Pro](/x/introduction/licensing/#pro-plan) plan), you need to update the import path:
```

**Review by LukasTy — APPROVED** [review]
🙌 💯

**Comment by oliviertassinari** [comment]
This closes #4250

**Comment by oliviertassinari** [comment]
Deprecation message added to the old npm package name:

SCR-20240917-slpt

https://www.npmjs.com/package/@mui/x-license-pro

**Pull request merged**